code/modules/projectiles/guns/energy/special.dm

```diff
@@ -305,15 +305,137 @@
 
 /obj/item/gun/energy/toxgun
 	name = "plasma pistol"
-	desc = "A specialized firearm designed to fire lethal bolts of toxins."
+	desc = "A specialized firearm designed to fire heated bolts of plasma. Can be overloaded for a high damage shield breaking shot."
 	icon_state = "toxgun"
-	fire_sound = 'sound/effects/stealthoff.ogg'
-
+	item_state = "toxgun"
+	sprite_sheets_inhand = list("Vox" = 'icons/mob/clothing/species/vox/held.dmi', "Drask" = 'icons/mob/clothing/species/drask/held.dmi') //This apperently exists, and I have the sprites so sure.
 	w_class = WEIGHT_CLASS_NORMAL
 	origin_tech = "combat=4;magnets=4;powerstorage=3"
-	ammo_type = list(/obj/item/ammo_casing/energy/toxplasma)
+	ammo_type = list(/obj/item/ammo_casing/energy/toxplasma, /obj/item/ammo_casing/energy/charged_plasma)
 	shaded_charge = 1
 	can_holster = TRUE
+	atom_say_verb = "beeps"
+	bubble_icon = "swarmer"
+	light_color = "#89078E"
+	light_power = 4
+	var/overloaded = FALSE
+	var/warned = FALSE
+	var/charging = FALSE
+	var/mob/living/carbon/holder = null
+
+/obj/item/gun/energy/toxgun/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/gun/energy/toxgun/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	UnregisterSignal(src, COMSIG_ITEM_DROPPED)
+	UnregisterSignal(holder, COMSIG_CARBON_SWAP_HANDS)
+	holder = null
+	return ..()
+
+/obj/item/gun/energy/toxgun/process() //keep your finger on the trigger soldier
+	..()
+	if(overloaded)
+		cell.charge -= 0.5 //2.5 per second, 25 every 10 seconds
+		if(cell.charge <= 25 && !warned)
```

Comment:
Best to use defines here instead of magic numbers. Same for the magic numbers below

Reply:
Done